book/sorting.md

```diff
@@ -0,0 +1,292 @@
+# Sorting
+
+Nushell offers many ways of sorting data, and which method you reach for will depend on the problem and what kind of data you're working with. Let's take a look at some of the ways you might wish to sort data.
+
+## Basic sorting
+
+### Lists
+
+Sorting a basic list works exactly how you might expect:
+
+```nu
+> [9 3 8 1 4 6] | sort
+╭───┬───╮
+│ 0 │ 1 │
+│ 1 │ 3 │
+│ 2 │ 4 │
+│ 3 │ 6 │
+│ 4 │ 8 │
+│ 5 │ 9 │
+╰───┴───╯
+```
+
+However, things get a bit more complex when you start combining types. For example, let's see what happens when we have a list with numbers _and_ strings:
+
+```nu
+> ["hello" 4 9 2 1 "foobar" 8 6] | sort
+╭───┬────────╮
+│ 0 │      1 │
+│ 1 │      2 │
+│ 2 │      4 │
+│ 3 │      6 │
+│ 4 │      8 │
+│ 5 │      9 │
+│ 6 │ foobar │
+│ 7 │ hello  │
+╰───┴────────╯
+```
+
+We can see that the numbers are sorted in order, and the strings are sorted to the end of the list, also in order. If you are coming from other programming languages, this may not be quite what you expect. In Nushell, as a general rule, **data can always be sorted without erroring**.
+
+::: tip
+If you _do_ want a sort containing differing types to error, see [strict sort](#strict-sort).
+:::
+
+Nushell's sort is also **stable**, meaning equal values will retain their original ordering relative to each other. This is illustrated here using the [insensitive](#insensitive-sort) sort option:
```

Comment:
Do we want to commit to all our sorting implementations being stable or should that be relaxed to `sort` or "stable by default"?

For clarity I would specify the insensitive
```suggestion
Nushell's sort is also **stable**, meaning equal values will retain their original ordering relative to each other. This is illustrated here using the [case insensitive](#insensitive-sort) sort option:
```

Reply:
`sort` and `sort-by` both use Rust's `sort_by`, which is stable. I also have a unit test which does something similar to the example here which demonstrates the sort being stable. I'm not sure what other sorting implementation there might be?

I agree, I'll rename the heading to "case insensitive" as well.